### ISSUE TITLE
Feature to create generic conditional sets for telemetry which can have fixed enum values.

### ISSUE THREAD

**Issue by bangpradyumna** [nasa/openmct#7685]
**Is your feature request related to a problem? Please describe.**
We have multiple telemetry of Switch type where the values can be either ON or OFF. It's frustrating to create a new conditional set for a every single switch telemetry (there are 100's). 

**Describe the solution you'd like**
I way to create conditional set which is not bound to any telemetry but rather is a generic one just dependent on the value of the telemetry. I would love to be able to just drag and drop a switch telemetry into panel and apply conditional formatting to it using this generic conditional set instead of having to create a new conditional set for it.

**Comment by bangpradyumna** [comment]
Please let me know if there already exists a way to accomplish this, I wasn't able to figure it out.

**Comment by jtrack3d** [comment]
A limit provider provides information based on a call from Provider.getLimitEvaluator() method.  

It returns "true" if the domain object supports "limits".

What if you wrote a generic plugin where supportsLimits returns true of domainObject.type == 'switch'.

Then, getLimitEvaluator() could return styling information you want for a switch.

It's sort of a limit concept.    I know I have tested using a limit provider on numerical data and if my styling returns colors, the labels change color in tables and labels.

**Comment by charlesh88** [comment]
@bangpradyumna Yes, you are absolutely right that our current structure creates a very time-consuming process for this use case. Because of similar needs at NASA, I built an online scripting tool that let's you generate Condition Sets, Widgets and related displays in bulk from spreadsheet CSV files. I've made layouts with hundreds of switchstate telem parameters that use conditionals to examine current state and style a widget green or gray, for example. Once you get your params and conditions defined in a spreadsheet, the tool runs in seconds and outputs a JSON file that is then imported into Open MCT using the "Import from JSON" feature.

The tool is freely available at my GitHub account at  https://charlesh88.github.io/openmct-scripting/. I've just pushed up a version that makes the usage documentation accessible in the tool at https://charlesh88.github.io/openmct-scripting/index-help.html. This includes a complete guide to setting up your input CSV files, and examples of same.

Please ping me back with _any_ questions about this - I hope you find it useful, and if so, I'd very much welcome feedback and comments.

**Comment by ozyx** [comment]
> @bangpradyumna I built an online scripting tool that let's you generate Condition Sets, Widgets and related displays in bulk from spreadsheet CSV files. I've made layouts with hundreds of switchstate telem parameters that use conditionals to examine current state and style a widget green or gray, for example. Once you get your params and conditions defined in a spreadsheet, the tool runs in seconds and outputs a JSON file that is then imported into Open MCT using the "Import from JSON" feature.


Sidenote, it would be awesome if we could turn this into a plugin 😎
